aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -422,9 +422,15 @@ void cpu_flash_attention(
             dst_data,
             headSize);
       }
+
       // dst <- dst / sum[row]
       // reorder MHA output with strides
       for (int64_t row = 0; row < qBlockSize; ++row) {
+        // Row sums for full masked out rows are 0, we set them to 1
+        // in order to avoid NaNs in the output and instead set fully
+        // masked out rows to 0
+        qk_max_data[row] = qk_max_data[row] == -std::numeric_limits<accum_t>::infinity() ? 0 : qk_max_data[row];
+        qk_sum_data[row] = qk_sum_data[row] == 0 ? 1 : qk_sum_data[row];
```

Comment:
dumb Q: why does setting the sums to 1 result in the output having 0s for fully masked-out rows?

Reply:
qk_sum_data is the the `sumexp(i-max)` When we set this. 

So  `qk_max_data[row] =  ..` This makes converts the output from nan to 0
and then 
`qk_sum_data[row]` This makes so that instead dividing by 0 on line 434 for normalizing we instead divide by 1